src/app/packages/open-layer-visualizers/annotations.visualizer.ts

```diff
@@ -0,0 +1,350 @@
+	public drawEndPublisher = new Subject();
+
+	// add special type for this one
+	public style: any = {
+		stroke: {
+			color: '#3399CC',
+			width: 2
+		},
+		fill: {
+			color: 'rgba(255,255,255,0.4)'
+		},
+		point: {
+			radius: 4
+		},
+		line: {
+			width: 2
+		}
+
+	};
+
+	constructor(style?: any){
+		super(AnnotationVisualizerType,[]);
+		if(style){
+			this.style = style;
+		}
+		this.geoJsonFormat = new GeoJSON();
+		this.features = [];
+		this.fill = true;
+	}
+
+	onInit(mapId: string, map: IMap){
+		this._imap = map;
+		this._mapId = mapId;
+	}
+
+	onResetView(){
+		this.addLayer();
+		this.redrawFromGeoJson();
+	}
+
+	getGeoJson() {
+		return JSON.stringify(this.getFeatures());
+	}
+
+	getFeatures(){
+		return featureCollection(this.features);
+	}
+
+	changeStroke(color) {
+		this.style.stroke.color = color;
+	}
+
+	changeFill(color) {
+		this.style.fill.color = color;
+	}
+
+	changeLine(width) {
+		this.style.stroke.width = width;
+	}
+
+	addLayer(id = this.type){
+		const layer = new VectorLayer();
+		//if id is empty then set the current type name as id
+		layer.set('id',id );
+		this.layer = this._imap.addLayerIfNotExist(layer);
+
+		this._source = new VectorSource({wrapX: false});
+		this.layer.setSource(this._source);
+		this.layer.setStyle(this.styleFunction.bind(this))
+		this.layer.setZIndex(200000)
+	}
+
+	removeLayer(){
+		this._imap.removeLayer(this.layer);
+	}
+
+	redrawFromGeoJson() {
+		//const geoFeatures = this.collection;
+		this._source.clear();
+		this.collection = undefined;
+		const features = this.createFeturesFromGeoJson(this.features);
+		this._source.addFeatures(features);
+	}
+
+	drawFeatures(data) {
+		if(data === undefined){
+			return ;
+		}
+
+		if(typeof data === 'string') {
+			data = JSON.parse(data);
+		}
+		// readFeatures throw exceptions so I am using this method
+
+		const features = this.createFeturesFromGeoJson(data.features);
+		this._source.addFeatures(features);
+	}
+
+	createFeturesFromGeoJson(geoJsonFeatures) {
+		const features = geoJsonFeatures.map((d) =>  this.geoJsonFormat.readFeature(d));
+
+		(<Array<any>>features).forEach( feature => {
+
+			const properties = feature.getProperties();
+			let geometry;
+
+			geometry = feature.getGeometry();
+			if (properties.geometryName === `${this.namePrefix}Circle`) {
+				geometry = new GeomCircle(geometry.getCoordinates(), properties.radius);
+			}
+			feature.setGeometryName(properties.geometryName);
+			feature.setGeometry(geometry);
+			feature.setStyle(this.styleFunction(feature, undefined, properties.style));
+		});
+
+		return features;
+	}
+
+	removeInteraction() {
+		if (this.interactionHandler) {
+			this._imap.removeInteraction(this.interactionHandler);
+		}
+	}
+
+	onDrawEndEvent(data) {
+		const geometryName = data.feature.getGeometryName();
+
+		data.feature.setStyle(this.styleFunction(data.feature, undefined));
+
+		data.feature.setProperties({
+			id: Date.now(),
+			style: this.style,
+			geometryName,
+			data: {}
```

Comment:
Add TODO: Don't send out reprojected original feature just new ones and prev original features